weave/trace/weave_client.py

```
@@ -1191,4 +1202,39 @@ def sanitize_object_name(name: str) -> str:
     return res
 
 
+# match local image file paths
+image_suffix = r".*\.(png|jpg|jpeg|gif|tiff)"
+image_pattern = re.compile(rf"^{image_suffix}$", re.IGNORECASE)
+remote_image_pattern = re.compile(rf"https://.*\.{image_suffix}")
+
+
+def convert_paths_to_images(obj: Any) -> Any:
+    """Convert paths to PathImage objects if they are image files."""
+    if isinstance(obj, dict):
+        for k, v in obj.items():
+            obj[k] = convert_paths_to_images(v)
+    elif isinstance(obj, list):
+        for v in obj:
+            convert_paths_to_images(v)
+    elif isinstance(obj, str):
+        if image_pattern.match(obj):
+            # Load the image, if PIL is available and file exists
+            try:
+                from PIL import Image
+
+                return PathImage(img=Image.open(obj), path=obj)
+            except (ImportError, FileNotFoundError) as e:
+                logger.warning(f"Failed to open image file: {obj}. {e}")
```

do we want to warn here? silently fail?